pygyro/advection/accelerated_advection_steps.py

```diff
@@ -223,95 +234,96 @@ def poloidal_advection_step_impl( f, dt, v, rPts, qPts,
                     #                               ^^^^^^^^^^^^^^^
-                    drPhi_k[i,j]     = eval_spline_2d_scalar(endPts_k1_q[i,j],endPts_k1_r[i,j],
-                                                        kts1Phi, deg1Phi, kts2Phi, deg2Phi,
-                                                        coeffsPhi,0,1)
-                    drPhi_k[i,j]     /= endPts_k1_r[i,j]
-                    dthetaPhi_k[i,j] = eval_spline_2d_scalar(endPts_k1_q[i,j],endPts_k1_r[i,j],
-                                                        kts1Phi, deg1Phi, kts2Phi, deg2Phi,
-                                                        coeffsPhi,1,0)
-                    dthetaPhi_k[i,j] /= endPts_k1_r[i,j]
+                    drPhi_k[i, j] = eval_spline_2d_scalar(endPts_k1_q[i, j], endPts_k1_r[i, j],
+                                                          kts1Phi, deg1Phi, kts2Phi, deg2Phi,
+                                                          coeffsPhi, 0, 1)
+                    drPhi_k[i, j] /= endPts_k1_r[i, j]
+                    dthetaPhi_k[i, j] = eval_spline_2d_scalar(endPts_k1_q[i, j], endPts_k1_r[i, j],
+                                                              kts1Phi, deg1Phi, kts2Phi, deg2Phi,
+                                                              coeffsPhi, 1, 0)
+                    dthetaPhi_k[i, j] /= endPts_k1_r[i, j]
                 else:
-                    drPhi_k[i,j]     = 0.0
-                    dthetaPhi_k[i,j] = 0.0
+                    drPhi_k[i, j] = 0.0
+                    dthetaPhi_k[i, j] = 0.0
 
                 # Step two of Heun method
                 # x^{n+1} = x^n + 0.5( f(x^n) + f(x^n + f(x^n)) )
                 # Clipping is one method of avoiding infinite loops due to
                 # boundary conditions
                 # Using the splines to extrapolate is not sufficient
-                endPts_k2_q[i,j] = (qPts[i] - (drPhi_0[i,j]     + drPhi_k[i,j])*multFactor) % (2*pi)
-                if endPts_k2_q[i,j]<0: # Needed for C due to pyccel issue #854
-                    endPts_k2_q[i,j] += 2*pi
-                endPts_k2_r[i,j] = rPts[j] + (dthetaPhi_0[i,j] + dthetaPhi_k[i,j])*multFactor
-
-                if (endPts_k2_r[i,j]<rPts[0]):
-                    endPts_k2_r[i,j]=rPts[0]
-                elif (endPts_k2_r[i,j]>rMax):
-                    endPts_k2_r[i,j]=rMax
-
-                diff=abs(endPts_k2_q[i,j]-endPts_k1_q[i,j])
-                if (diff>norm):
-                    norm=diff
-                diff=abs(endPts_k2_r[i,j]-endPts_k1_r[i,j])
-                if (diff>norm):
-                    norm=diff
-                endPts_k1_q[i,j]=endPts_k2_q[i,j]
-                endPts_k1_r[i,j]=endPts_k2_r[i,j]
+                endPts_k2_q[i, j] = (
+                    qPts[i] - (drPhi_0[i, j] + drPhi_k[i, j])*multFactor) % (2*pi)
+                if endPts_k2_q[i, j] < 0:  # Needed for C due to pyccel issue #854
+                    endPts_k2_q[i, j] += 2*pi
```

Comment:
Modulo operator now works

Reply:
The problem was that the modulo operator was not implemented correctly; now these lines are in general unnecessary.

Reply:
It seems there is a bug with `%=` you will need `a = a%(2*pi)`

Reply:
I have opened an issue in pyccel https://github.com/pyccel/pyccel/issues/1172

Reply:
Great, I replaced it by `%(2*pi)`